update search (#23)
SenseNet/docs.sensenet.com: content/concepts/content-management/04-content-templates.md

After selecting a content template the user will be presented a content view for that content type and will be able to see and change the predefined template fields (depending on the content view). When the user presses the Done editing button the whole template will be copied to the destination place. E.g. if the template is of the workspace type, all its lists and libraries will be present at the destination place.

See example under API docs [https://docs.sensenet.com/api-docs/content-management/01-create]

# Permissions and security

Content templates can be controlled by permissions the same way as any other content. If the user has no permission for a particular template she will not be able to create content using that template.
After creating a content using a template the security settings that were set directly on the content template are also copied to the new content. All settings set on the destination parent will be inherited by the new content naturally.

> 

Author: ```suggestion
To build a separate micro permission system locally it is enough to add local groups to wokspace templates.
```

# Creating a template

Content templates are content of the same type as they are defined for. For example document templates are files; library or list templates are document libraries or memo lists. To create a content template you simply need to create a new content under the appropriate folder (named by the type). You can set any property of the content template or add child content (e.g. custom views for libraries), they will be part of the template.

# Modify or deleting a template

You can modify or delete a content template at any time. They are not connected to the content created using them in any way. For this reason, modifying a template will not modify a content previously created.

# Creating a new content with a template

Content templates are recognized and cached by the system immediately. You can create new content using the templates in any folder where that particular type is allowed.

> When you open a New menu anywhere it will contain the following items for every allowed type:

Reviewer: unfortunately this is not working this way anymore, so this paragraph with the new menu could be removed

- if the content type has no content template, the type name will be listed (e.g. New Memo)
- in case there are existing templates for that type all relevant templates will be listed (for example if the File content type is allowed in a document library, all the document templates will appear in the New menu with their name displayed)

> After selecting a content template the user will be presented a content view for that content type and will be able to see and change the predefined template fields (depending on the content view). When the user presses the Done editing button the whole template will be copied to the destination place. E.g. if the template is of the workspace type, all its lists and libraries will be present at the destination place.

Reviewer: this part should be reduced to only summarize the actual creation process briefly (under the hood)

See example under API docs [https://docs.sensenet.com/api-docs/content-management/01-create]

# Permissions and security

Content templates can be controlled by permissions the same way as any other content. If the user has no permission for a particular template she will not be able to create content using that template.
After creating a content using a template the security settings that were set directly on the content template are also copied to the new content. All settings set on the destination parent will be inherited by the new content naturally.

> 

Reviewer: local groups can be also added to workspace templates so this way the user is able to build separate micro permission system locally

# Content template hierarchy

Content templates can be defined on the following levels:

- global: global folder, see above

> - site: ContentTemplates system folder under a site

Reviewer: ```suggestion
```

- workspace: ContentTemplates system folder under a workspace
- local: ContentTemplates system folder under any content - e.g. document library

> When the user opens a new menu the system looks for content templates defined for all content types that are allowed there. It starts under the current content (e.g. content list), than moves up to the workspace, site and finally the global level. Content templates can override each other by name. This means if you define a local content template with a name that exists above on the workspace, site or global level, it will override them.

Reviewer: ```suggestion
When the user creates new content the system looks for content templates defined for all content types that are allowed there. It starts under the current content (e.g. content list), than moves up to the workspace and finally the global level. Content templates can override each other by name. This means if you define a local content template with a name that exists above on the workspace, site or global level, it will override them.
```
